exercises/binary-search/src/example/java/BinarySearch.java

```diff
@@ -0,0 +1,70 @@
+
+import java.util.List;
+
+public class BinarySearch<T extends Comparable<T>> {
+
+    public static String ARRAY_MUST_BE_SORTED = "Array should be sorted.";
```

Comment:
In general, the code should be self-explanatory will little need for comment.

If one is tempted to add a comment, ask first, "is there a way I can reword this code to make it so clear, a comment is no longer necessary?"

Comments are best used to explain unusual/surprising context... that is to say, they share *why* a chunk of code is shaped in the way it is.

For example, in `robot-simulator`, @stkent smartly overloaded `Object::equals` instead of overriding it and didn't implement `Object::hashCode`.  He left a comment explaining why and how one would do it in the real world.